smack-extensions/src/main/java/org/jivesoftware/smackx/xdata/provider/DataFormProvider.java

```diff
@@ -210,19 +210,22 @@ private static FormField parseField(XmlPullParser parser, XmlEnvironment xmlEnvi
             // The field name 'FORM_TYPE' is magic.
             if (fieldName.equals(FormField.FORM_TYPE)) {
                 type = FormField.Type.hidden;
-            } else {
+            } else if (formType != null) {
                 // If no field type was explicitly provided, then we need to lookup the
                 // field's type in the registry.
                 type = FormFieldRegistry.lookup(formType, fieldName);
                 if (type == null) {
                     LOGGER.warning("The Field '" + fieldName + "' from FORM_TYPE '" + formType
                                     + "' is not registered. Field type is unknown, assuming text-single.");
-                    // As per XEP-0004, text-single is the default form field type, which we use as emergency fallback here.
-                    type = FormField.Type.text_single;
                 }
             }
         }
 
+        if (type == null) {
```

Comment:
That should be a `else` branch in the `if/else` above

Reply:
Committed change to make it so. However it does result in a small duplication of the code setting the default. An alternative would be the replacing the `else` with another `if (type == null)`, but inside the initial `if (type == null)` block. That would remove the duplication, but prevent a double `(type == null)` check when a type was set in the first place.